snapshots/helpers_test.go

```diff
@@ -100,7 +101,10 @@ func (m *mockSnapshotter) Snapshot(height uint64, format uint32) (<-chan io.Read
 // setupBusyManager creates a manager with an empty store that is busy creating a snapshot at height 1.
 // The snapshot will complete when the returned closer is called.
 func setupBusyManager(t *testing.T) *snapshots.Manager {
-	tempdir := t.TempDir()
+	tempdir, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = os.RemoveAll(tempdir) })
```

Comment:
why this works, and `t.TempDir` doesn't?

Reply:
Take a look at how testing.T.TempDir() is implemented. Plus, please read the original commit's comment. It's quite a common issue

Reply:
OK, thanks. Maybe we should add a comment to the code explaining shortly why we use `ioutil.TempDir` or link the golang issue?

Reply:
Yeah I thought about it. Then I thought that `git blame` & `git log` may provided enough documentation. But you're right, an inline comment would be definitely more helpful. I'll fix it in a PR to master

Reply:
it takes way more energy to do `git blame; git log` than adding a comment. Also, it's much easier to read that comment then think about this workaround.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/8508